client/shared/src/search/query/decoratedToken.ts

```diff
@@ -431,6 +517,40 @@ export const hasRegexpValue = (field: string): boolean => {
     }
 }
 
+const specifiesRevision = (value: string): boolean => value.match(/@/) !== null
+
+const decorateRepoRevision = (token: Literal): DecoratedToken[] => {
+    const [repo, revision] = token.value.split('@', 2)
+    const offset = token.range.start
+
+    return Array.prototype.concat(
+        decorate({
+            type: 'pattern',
+            kind: PatternKind.Regexp,
+            value: repo,
+            range: { start: offset, end: offset + repo.length },
+        }),
+        [
+            {
+                type: 'metaRepoRevisionSeparator',
+                value: '@',
+                range: {
+                    start: offset + repo.length,
+                    end: offset + repo.length + 1,
+                },
+            },
+        ],
+        mapRevisionMeta({
+            type: 'literal',
+            value: revision,
+            range: {
+                start: token.range.start + repo.length + 1,
+                end: token.range.start + repo.length + 1 + revision.length,
+            },
+        })
+    ) as DecoratedToken[]
```

Comment:
```suggestion
    return [
        ...decorate({
            type: 'pattern',
            kind: PatternKind.Regexp,
            value: repo,
            range: { start: offset, end: offset + repo.length },
        }),
        {
            type: 'metaRepoRevisionSeparator',
            value: '@',
            range: {
                start: offset + repo.length,
                end: offset + repo.length + 1,
            },
        },
        ...mapRevisionMeta({
            type: 'literal',
            value: revision,
            range: {
                start: token.range.start + repo.length + 1,
                end: token.range.start + repo.length + 1 + revision.length,
            },
        })
    ]
```

<img src="https://media.giphy.com/media/f9wiV1QYsy2tRp7mDP/giphy.gif">

Reply:
![](https://media4.giphy.com/media/MPLpvJcsWvrkk/200.gif?cid=5a38a5a25dhj2tddyzhsr4tvan69m8shwr29gmifcst68ek1&rid=200.gif)